### ISSUE TITLE
TCP.onread block when working with readline on v0.12.2 OSX

### ISSUE THREAD

**Issue by rain1017** [joyent/node#25492]
I make one tcp request when user input a line, but the TCP response is usually block until making next a few keystrokes. Tcpdump shows that the response packet has reached the client, but TCP.onread does not fired.
I use node v0.12.2, this is only happen in Mac OSX.  Ubuntu is OK

echo server
```js
var net = require('net');

var server = net.createServer(function(socket){
    socket.on('data', function(msg){
        console.log('%s', msg);
        socket.write(msg);
    });
});

server.listen(31017, '127.0.0.1');
```

client
```js
console.log('Type anything then press enter, you should get response from echoServer immediately');
console.log('Sometimes TCP.onread may block until you make next keystroke');

var net = require('net');
var rl = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal : true,
});

var socket = net.createConnection({port : 31017});
socket.on('data', function(buf){
    console.log('=> %s', buf);
});

rl.on('line', function(line){
    socket.write(line);
})
.on('close', function(){
    socket.end();
    process.exit(0);
});
```

**Comment by rain1017** [comment]
I believe is a new bug in Node v0.12 on OSX.  
Tested v0.10.36 on OSX, not repo.
v0.12 on Ubuntu, not repo.

**Comment by misterdjules** [comment]
Thank you @rain1017 for the detailed report! Confirmed bug, fixed by https://github.com/joyent/node/pull/25475. The fix should be included in the next release, node v0.12.5.

**Comment by misterdjules** [comment]
Closing as a duplicate of https://github.com/joyent/node/issues/9310.
